app/lib/reportback/index.js

```diff
@@ -351,26 +328,47 @@ function createUserThenReportBack(doc, data) {
  *   Data from the user's request
  */
 function submitReportBack(uid, doc, data) {
-  var rbData = {
-    nid: data.campaignConfig.campaign_nid,
-    uid: uid,
-    caption: doc.caption,
-    quantity: doc.quantity,
-    why_participated: doc.why_important,
-    file_url: doc.photo
-  };
+  var data = data
+    , customFields = {}
+    , rbData = {
+        nid: data.campaignConfig.campaign_nid,
+        uid: uid,
+        caption: doc.caption,
+        quantity: doc.quantity,
+        why_participated: doc.why_important,
+        file_url: doc.photo
+      }
+    ;
 
   dscontentapi.campaignsReportback(rbData, function(err, response, body) {
     if (err) {
       logger.error(err);
     }
     else if (body && body.length > 0) {
-      if (body[0] == false) {
+      var rbId = body[0]
+      // Checking to make sure the response body doesn't contain an error from the API. 
+      if (rbId == false || isNaN(rbId)) {
```

Comment:
Sometimes the API gives a descriptive error string in the response body **instead** of an error. I've received something like "you need to authenticate first" as the element in the body array, instead of an error from the API, strangely enough.